pkg/v1/mutate/mutate.go

```diff
@@ -113,6 +113,29 @@ func Config(base v1.Image, cfg v1.Config) (v1.Image, error) {
 	return ConfigFile(base, cf)
 }
 
+// ManifestAnnotations mutates the provided v1.Image to have the provided annotations
+func ManifestAnnotations(base v1.Image, annotations map[string]string) (v1.Image, error) {
+	m, err := base.Manifest()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Can we `m.DeepCopy` here instead, as early as possible and before any changes?

Reply:
thank you so much for this review, but I tried to do that but it didn't work for me, I updated the code with the following content, what did I do wrong?
```go
mn := m.DeepCopy()

if mn.Annotations == nil {
	mn.Annotations = map[string]string{}
}

for k, v := range annotations {
	mn.Annotations[k] = v
}

image := &image{
	base:     base,
	manifest: mn,
}
```